### DISCUSSION TITLE
Cached page not not hitting middleware

### DISCUSSION THREAD

**Discussion by MerlinMason** [vercel/next.js#46608] [Help]
For our setup we use client side data fetching (Tanstack query) and middleware which handles authentication with JWTs. 

On initial page load we get a `200` response and everything works fine, the middleware is hit and the user is authenticated. 

On subsequent page loads we get a `304` response (not modified) and the middleware is not hit. 

This is not a problem **unless** the JWT has expired, in this case the user doesn't have a valid session and as the middleware is not invoked we cannot use the refresh token to get a fresh JWT.

What's the best way to tell NextJS to always invoke the middleware?

From experimenting, adding an empty `getServerSideProps` seems to work:

```js
export const getServerSideProps: GetServerSideProps = async () => ({
    props: {},
});
```

However as our entire site requires authentication I don't really want to add this to every page, is there a better way?

There's a few parts of the docs which are slightly misleading here too:
> _Edge Middleware runs on every request by default_

and

> _Middleware runs before cached content_

This does not seem to be the case, are the docs wrong? Is this a bug?

#### Context
- Next 13.1.1
- Hosted on Vercel
- `/pages` directory

**Comment by icyJoseph** [comment]
> On subsequent page loads we get a 304 response (not modified) and the middleware is not hit.

Could you check if there's an `X-VERCEL-` something like that header on the response? I think this might be something with the CDN rather... Like the request is not even arriving to the Next.js server.

It is called X-Vercel-Cache

**Reply by MerlinMason** [reply]
Hey, thanks for the response! 

Yes there's a `x-vercel-cache:` header - when getting a `200` it's value is `MISS` when getting a `304` it's value is `HIT`.

**Comment by Kethatril** [comment]
This is also happening to us, middleware does not seem to run on a 304 response and it always runs on a 200 response regardless of whether `x-vercel-cache` is `HIT` `MISS` or `STALE`
